soccer/KickEvaluator.cpp

```diff
@@ -0,0 +1,396 @@
+#include "KickEvaluator.hpp"
+#include <Utils.hpp>
+#include <Geometry2d/Util.hpp>
+
+#include <algorithm>
+#include <vector>
+#include <math.h>
+#include <cmath>
+
+static union {
+    float d;
+    struct {
+#ifdef LITTLE_ENDIAN
+        int j, i;
+#else
+        int i, j;
+#endif
+    } n;
+} _eco;
+
+// Floating point magic
+#define EXP_A (1048576 / 0.69314718055994530942)
+#define EXP_C 60801
+#define fast_exp(y) (_eco.n.i = EXP_A * (y) + (1072693248 - EXP_C), _eco.d)
```

Comment:
Yeah taylor series will be alot slower. Thats why i made the comment that I was being stupid =D. That was optimized for running in compile time on literals. This one should have the same or better performance since it looks basically the same as the macro.
http://www.smipple.net/snippet/Martin%20Ankerl/A%20Fast,%20Compact%20Approximation%20of%20the%20Exponential%20Function